src/magic_folder/create.py

```diff
@@ -123,31 +39,42 @@ def magic_folder_create(alias, nickname, name, node_directory, local_dir, poll_i
     :param HTTPClient treq: An ``HTTPClient`` or similar object to use to make
         the queries.
 
-    :return Deferred[integer]: A status code of 0 for a successful execution. Otherwise
-        an appropriate exception is raised.
+    :return Deferred: ``None`` or an appropriate exception is raised.
     """
-    folders = load_magic_folders(node_directory)
-
-    # make sure we don't already have a magic-folder with this name before we create the alias
-    if name in folders:
-        raise Exception("Already have a magic-folder named '{}'".format(name))
 
-    rc = yield tahoe_create_alias(node_directory, alias, treq)
-    if rc != 0:
-        raise Exception("Failed to create alias")
+    # XXX probably want to pass this in, instead of "treq"?
+    tahoe_client = create_tahoe_client(config.tahoe_client_url, treq)
 
-    if nickname is not None:
-        # inviting itself as a client
-        from twisted.internet import reactor
-        treq = HTTPClient(Agent(reactor))
-
-        try:
-            invite_code = yield _invite(node_directory, alias, nickname, treq)
-        except Exception as e:
-            raise Exception("Failed to invite after create: {}".format(str(e)))
-
-        rc = _join(invite_code, node_directory, local_dir, name, poll_interval, nickname)
-        if rc != 0:
-            raise Exception("Failed to join after create")
+    if name in config.list_magic_folders():
+        raise Exception("Already have a magic-folder named '{}'".format(name))
 
-    returnValue(0)
+    # create our author
+    author = create_local_author(author_name)
+
+    # create an unlinked directory and get the dmd write-cap
+    collective_write_cap = yield tahoe_client.create_mutable_directory()
+
+    # create the personal dmd write-cap
+    personal_write_cap = yield tahoe_client.create_mutable_directory()
```

Comment:
Don't we need to derive a read-only cap of our personal write cap and then pass that as `entry_cap`?

Reply:
Yes! Great catch :)